examples/java-calculator/src/test/java/io/cucumber/examples/java/ShoppingSteps.java

```diff
@@ -19,9 +45,9 @@ public void the_following_groceries(List<Grocery> groceries) {
         }
     }
 
-    @When("I pay {}")
-    public void i_pay(int amount) {
-        calc.push(amount);
+    @When("I pay {price}")
+    public void i_pay(Price amount) {
```

Comment:
I think the object should be called `Amount`. Perhaps we can make this example a bit more powerful by making `Amount` a tuple of `Currency` and `BigDecimal`.